pyccel/codegen/printing/ccode.py

```diff
@@ -286,19 +286,27 @@ def _print_FunctionDef(self, expr):
                     decs = decs,
                     body = body))
 
+    def stored_in_c_pointer(self, a):
+        if not isinstance(a, Variable):
+            return False
+        return a.is_pointer or a.is_optional
+
     def _print_FunctionCall(self, expr):
         func = expr.funcdef
+         # Ensure the correct syntax is used for pointers
+        args = [VariableAddress(a) if self.stored_in_c_pointer(a) else a for a in expr.arguments]
```

Comment:
`VariableAddress` is not needed for every argument. It should only be needed occasionally.
Eg if the function signature is
`void f(int a)`
Then we need to print `Variable`s in the function call not `VariableAddress`
However if the function call is
`void f(int* a)`
Then we need to pass a `VariableAddress`

Reply:
Whether you pass `VariableAddress` or not depends on the signature not the type that you are passing
Suppose you have the following variables:
```
int a;
int* p;
```
We need variable when we should use `a` or `*p`, and `VariableAddress` where we should use `&a` or `p`. If the 2 are not interchangeable then you are either in an `AliasAssign` or you're probably doing something wrong

If the function being called is:
`void f(int c)`
Then the function call should be:
`f(a);`
or
`f(*p);`
Similarly if the function is:
`void g(int* d)`
Then the function call should be:
`g(&a);`
Or
`g(p);`
Currently you are doing:
`f(a)`, `g(a)`, `f(p)`, `g(p)`